flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/node/DataMapperBuilder.java

```diff
@@ -155,34 +138,38 @@ public void setConcreteTemplateData(TemplateContext context) {
                     .stepOut()
                 .type(Property.ValueType.SINGLE_SELECT)
                 .value("")
-                .typeConstraint(new ArrayList<>(visibleRecordTypes))
+                .typeConstraint(new ArrayList<>(visibleTypes))
                 .optional(false)
                 .editable()
                 .stepOut()
                 .addProperty(OUTPUT_KEY);
     }
 
-    private static Optional<String> getVariableSignature(SemanticModel semanticModel, ModuleInfo moduleInfo,
-                                                         String name, TypeSymbol typeSymbol) {
-        if (name == null || name.isEmpty()) {
-            return Optional.empty();
-        }
-        String typeSignature =
-                CommonUtils.getTypeSignature(semanticModel, typeSymbol, false, moduleInfo);
-        return Optional.of(typeSignature + " " + name);
-    }
-
-    private static Optional<String> getRecordTypeSignature(TypeDefinitionSymbol symbol) {
-        if (symbol.typeDescriptor().typeKind() != TypeDescKind.RECORD) {
-            return Optional.empty();
-        }
-        Optional<String> moduleName = symbol.getModule().flatMap(Symbol::getName);
-
-        // TODO: Make this more scalable
-        if (moduleName.isPresent() && moduleName.get().equals("lang.annotations")) {
-            return Optional.empty();
+    private void addDataMappingCapableTypes(Set<String> types, Symbol parentSymbol, TypeSymbol typeSymbol) {
+        TypeSymbol rawType = CommonUtils.getRawType(typeSymbol);
+        switch (rawType.typeKind()) {
+            case ARRAY:
+                addDataMappingCapableTypes(types, parentSymbol, ((ArrayTypeSymbol) rawType).memberTypeDescriptor());
+                break;
+            case RECORD:
+                Optional<String> moduleName = parentSymbol.getModule().flatMap(Symbol::getName);
+                if (moduleName.isPresent() && moduleName.get().equals("lang.annotations")) {
+                    break;
+                }
+                parentSymbol.getName().ifPresent(types::add);
+                break;
+            case NIL:
+            case BOOLEAN:
+            case INT:
+            case FLOAT:
+            case DECIMAL:
+            case BYTE:
+            case STRING:
+            case JSON:
+                parentSymbol.getName().ifPresent(types::add);
+                break;
+            default:
```

Comment:
Shall we replace this with an [enhanced switch statement](https://www.jetbrains.com/help/inspectopedia/EnhancedSwitchMigration.html)?

Reply:
Fixed